microsoft/testsuites/dpdk/testpmd.py

```diff
@@ -0,0 +1,155 @@
+    ]
+
+    _redhat_packages = [
+        "gcc",
+        "make",
+        "git",
+        "tar",
+        "wget",
+        "dos2unix",
+        "psmisc",
+        "kernel-devel",
+        "numactl-devel.x86_64",
+        "librdmacm-devel",
+        "pkgconfig",
+        "libmnl-devel",
+        "elfutils-libelf-devel",
+        "python3-pip",
+    ]
+    _rte_target = "x86_64-native-linuxapp-gcc"
+    _dpdk_github = "https://github.com/DPDK/dpdk.git"
+    _ninja_url = (
+        "https://github.com/ninja-build/ninja/releases/download/v1.10.2/ninja-linux.zip"
+    )
+
+    def __execute_assert_zero(self, cmd: str, path: PurePath) -> str:
+        result = self.node.execute(cmd, sudo=True, shell=True, cwd=path)
+        assert_that(result.exit_code).is_zero()
+        return result.stdout
+
+    @property
+    def can_install(self) -> bool:
+        for _os in [Ubuntu, CentOs, Redhat]:
+            if isinstance(self.node, _os):
+                return True
+        return False
+
+    @property
+    def dependencies(self) -> List[Type[Tool]]:
+        return [Git, Wget]
+
+    def _install(self) -> bool:
+        if self._install_dependencies():
+            node = self.node
+            git_tool = node.tools[Git]
+            git_tool.clone(self._dpdk_github, cwd=node.working_path)
+            dpdk_path = self.node.working_path.joinpath("dpdk")
+            self.__execute_assert_zero("meson build", dpdk_path)
+            dpdk_build_path = dpdk_path.joinpath("build")
+            self.__execute_assert_zero("which ninja", dpdk_build_path)
+            self.__execute_assert_zero("ninja", dpdk_build_path)
+            self.__execute_assert_zero("ninja install", dpdk_build_path)
+            self.__execute_assert_zero("ldconfig", dpdk_build_path)
+            return True
+        else:
+            return False
+
+    def _install_dependencies(self) -> bool:
+        node = self.node
+        cwd = node.working_path
+        if isinstance(node.os, Ubuntu):
+            for package in self._ubuntu_packages:
+                node.os.install_packages(package)
+            self.__execute_assert_zero("pip3 install --upgrade meson", cwd)
+            self.__execute_assert_zero("mv /usr/bin/meson /usr/bin/meson.bak", cwd)
+            self.__execute_assert_zero("ln -s /usr/local/bin/meson /usr/bin/meson", cwd)
+            self.__execute_assert_zero("pip3 install --upgrade ninja", cwd)
+            self.__execute_assert_zero(
+                "modprobe -a rdma_cm rdma_ucm ib_core ib_uverbs mlx4_core mlx4_ib", cwd
+            )
+
+        elif isinstance(node.os, Redhat) or isinstance(node.os, CentOs):
+            self.__execute_assert_zero(
+                "yum update -y --disablerepo='*' --enablerepo='*microsoft*'", cwd
+            )
+            node.execute(
+                "yum groupinstall -y 'Infiniband Support'",
+                sudo=True,
+                shell=True,
+            )
+            for package in self._redhat_packages:
```

Comment:
The packages can be installed by the list, that would be more efficient than one by one.

Reply:
Ah, I wanted it to error out on a specific package if there was a problem installing one of them. I can swap it out to use the list now that I've sorted out which ones are needed.

Reply:
The log of os is optimized, but may not good enough. It should be able to show which package is failed to install. If it doesn't work as expected, please help on fixing it.